library/pk.c

```diff
@@ -791,6 +815,11 @@ int mbedtls_pk_check_pair(const mbedtls_pk_context *pub,
         }
     }
 
+#if defined(MBEDTLS_USE_PSA_CRYPTO)
+    if (pub->pk_info->type != MBEDTLS_PK_RSA) {
+        return prv->pk_info->check_pair_func(pub, prv, f_rng, p_rng);
+    } else
+#endif
```

Comment:
As in other places comments are added. I suggest to add also here.
```suggestion
#endif /* MBEDTLS_USE_PSA_CRYPTO */
```